google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryImpl.java

```diff
@@ -445,46 +446,72 @@ public TableDataInsertAllResponse call() throws Exception {
   }
 
   @Override
-  public Page<FieldValueList> listTableData(String datasetId, String tableId,
-      TableDataListOption... options) {
-    return listTableData(TableId.of(datasetId, tableId), getOptions(), optionMap(options));
+  public TableResult listTableData(
+      String datasetId, String tableId, TableDataListOption... options) {
+    return listTableData(TableId.of(datasetId, tableId), options);
   }
 
   @Override
-  public Page<FieldValueList> listTableData(TableId tableId, TableDataListOption... options) {
-    return listTableData(tableId, getOptions(), optionMap(options));
+  public TableResult listTableData(TableId tableId, TableDataListOption... options) {
+    return listTableData(tableId, null, options);
```

Comment:
I'm having trouble following which overloaded method this is going to. If it's the one above, it seems to be using `tableId` as the `datasetId`.

I suppose it's intended that the Schema is null, but I'm not sure how Java knows that.

Also, what was BigQueryOptions for and why is it going away?

Reply:
This is calling overload on line 466.
The first argument has type `TableId` which disqualifies L449 and L460, so L466 is the only one left with the right number of args.

The `BigQueryOptions` isn't going away, it's just being populated on L468.

I think this means we have too many overloads, but I'm not sure if we should require the user to pass null schema.

Reply:
Thanks for clarifying. Yeah, too many overloads. But we shouldn't force them to explicitly pass in null when they don't know the schema.